src/main/flight/imu.c

```diff
@@ -529,3 +526,73 @@ void imuSetHasNewData(uint32_t dt)
 }
 #endif
 
+void imuQuaternionComputeProducts(quaternion *quat) {
+    quat->ww = quat->w * quat->w;
+    quat->wx = quat->w * quat->x;
+    quat->wy = quat->w * quat->y;
+    quat->wz = quat->w * quat->z;
+    quat->xx = quat->x * quat->x;
+    quat->xy = quat->x * quat->y;
+    quat->xz = quat->x * quat->z;
+    quat->yy = quat->y * quat->y;
+    quat->yz = quat->y * quat->z;
+    quat->zz = quat->z * quat->z;
+}
+
+bool imuQuaternionHeadfreeDislocationSet(void) {
```

Comment:
@adrianmiriuta:

> it is a foreign word in english , and comes from the latin "dislocare"
and means "shift" , "adjustment", "displacement".

I see where you are going wrong there. The fact that a word has been borrowed from another language does **not** mean that it has the same meaning in the new language that it had in the language it has been borrowed from.

> https://en.oxforddictionaries.com/definition/dislocation

This (or Merriam-Webster for the USA) is pretty much the 'official' reference for the English language. And according to it, the meanings you attribute to 'dislocation' do not exist.